src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlCommand.cs

```diff
@@ -467,25 +456,55 @@ internal SqlStatistics Statistics
         {
             get
             {
-                // if the transaction object has been zombied, just return null
-                if ((null != _transaction) && (null == _transaction.Connection))
-                {
-                    _transaction = null;
-                }
-                return _transaction;
+                // there is no real reason to read this value except when creating a new command based on this command
+                // so we just use the Connection's CurrentTransaction since that it the only valid value anyway
+                SqlTransaction transaction = (_activeConnection?.InnerConnection as SqlInternalConnectionTds)?.CurrentTransaction?.Parent;
+                return transaction?.IsZombied == true ? default(SqlTransaction) : transaction;
             }
             set
             {
                 // Don't allow the transaction to be changed while in an async operation.
-                if (_transaction != value && _activeConnection != null)
+                if (Transaction != value && _activeConnection != null)
                 { // If new value...
                     if (cachedAsyncState.PendingAsyncOperation)
                     { // If in pending async state, throw
                         throw SQL.CannotModifyPropertyAsyncOperationInProgress();
                     }
                 }
 
-                _transaction = value;
+                // on null transaction just move on
+                if (value is null)
+                {
+                    return;
+                }
```

Comment:
I understand that a null value here wouldn't matter as the command will just use the connection's transaction. But since this setter is validating the input, shouldn't null also throw? A user assigning null to this Transaction property will think the command will execute under that transaction, which they believe to be null.